anesthetic/plot.py

```diff
@@ -281,6 +288,16 @@ def fastkde_plot_1d(ax, data, *args, **kwargs):
 
     ans = ax.plot(x[i], p[i], color=color, *args, **kwargs)
     ax.set_xlim(*check_bounds(x[i], xmin, xmax), auto=True)
+
+    if facecolor is True:
+        facecolor = color
+        c = iso_probability_contours(p[i], contours=levels)
```

Comment:
This logic isn't quite right. It should be
```python
if facecolor:
    if facecolor is True:
        facecolor = color
    c = ...
```
The difference being that if you pass `facecolor='blue'` as an argument, then this would give you control over the contour colouring. It's unlikely you'd actually want this except for when you want dark edges by passing `facecolor='blue', color='k'` as kwargs.

The same should be done in the second function

Reply:
If the logic is as you suggest then when facecolor is none and cmap is also none the following

```python
if cmap is None:
        cmap = basic_cmap(facecolor)
```

will throw an error.

For it to work as you suggest you would need to remove the cmap kwarg probably and add in an edgecolor=color to fill_between. I think this would work well...

```python
if facecolor:
        if facecolor is True:
            facecolor = color
        c = iso_probability_contours(p[i], contours=levels)
        cmap = basic_cmap(facecolor)
        for j in range(len(c)-1):
            ax.fill_between(x[i], p[i], where=p[i] >= c[j],
                            color=cmap(c[j]), edgecolor=color)
```

Reply:
@williamjameshandley This set up actually gives the user quite a bit of freedom with the colour scheme e.g.

```python
from anesthetic.samples import MCMCSamples
import matplotlib.pyplot as plt
import numpy as np

mu1 = [2, 10, -1]
mu2 = [0, 8, 0]
cov1 = 1*np.eye(3)
cov2 =  1*np.eye(3)

A = MCMCSamples(np.vstack([
    np.random.multivariate_normal(mu1, cov1, 2000),
    np.random.multivariate_normal(mu2, cov2, 2000)]))

fig, ax = A.plot_2d(
    [0, 1],
    types={'diagonal': 'kde', 'lower': 'kde', 'upper': 'scatter'},
    diagonal_kwargs={'facecolor': 'C0', 'color': 'k'})
plt.show()
```
Looks pretty nice to me...
![plot_example](https://user-images.githubusercontent.com/40355093/110383497-464d2080-8054-11eb-852f-28759c368bcf.png)

Reply:
Nice!